plugin-gradle/src/main/java/com/diffplug/gradle/spotless/SpotlessTask.java

```diff
@@ -177,10 +185,8 @@ public void performAction(IncrementalTaskInputs inputs) throws Exception {
 				.build()) {
 			// determine if a list of files has been passed in
 			Predicate<File> shouldInclude;
-			Project project = getProject();
-			if (project.hasProperty(FILES_PROPERTY) && project.property(FILES_PROPERTY) instanceof String) {
-				Object rawIncludePatterns = project.property(FILES_PROPERTY);
-				final String[] includePatterns = ((String) rawIncludePatterns).split(",");
+			if (this.filePatterns != null && !this.filePatterns.isEmpty()) {
```

Comment:
No need for a defensive `if != null`, since you're initializing to empty and have Objects.requireNonNull on the setter.

Reply:
removed